### PR TITLE
Plugins Update Checker and Refactor Plugins Information

### PR DESCRIPTION
**Pull request by Tmeister**
# Pull Request

## What changed?

- Added the Plugin Update Check endpoint
- Added Form Request for validation and rules
- Added Services to encapsulate logic
- Added BasePluginResource to avoid duplicate code
- Added Test for Plugin Information pagination

## Why did it change?

Refactor: Keep the code clean, avoid duplicate code, and follow "Laravel's way."

To add the logic to the endpoint `/plugins/update-check/1.1/`

## Did you fix any specific issues?

Related: #56 #57

## CERTIFICATION

By opening this pull request, I do agree to abide by
the [CODE OF CONDUCT](https://github.com/aspirepress/.github/CODE_OF_CONDUCT.md) and be bound by the terms
of the [Contribution Guidelines](https://github.com/aspirepress/.github/CONTRIBUTING.md) in effect on the date and time
of my contribution as proven by the
revision information in GitHub. I also agree that any previous contributions shall be deemed subject to the terms of the
version in effect on the date and time of this pull request, or any future revisions for pull requests I may submit.
Further, I certify that this work is my own, is original, does not violate the intellectual property of any other person
or entity, and I am not violating any license agreements or contracts I have with any person or entity. Finally, I agree
that this code may be licensed under any license deemed appropraite by AspirePress, including but not
limited to open source, closed source, proprietary or custom licenses, and that such license terms neither violate my
rights or my copyright to this code.

### CODE REVIEW

**Review by chuckadams — CHANGES_REQUESTED** [review]
The FormData classes are something we'll definitely want to move to laravel-data, but I'm not going to let that hold up new code.  Only change that needs to happen is for PluginUpdateCheckTest to move to bruno or be in a separate test suite that's disabled by default.

File: tests/Feature/API/WpOrg/PluginUpdateCheckTest.php
[the remark this answers is not in the document]
Comment (Tmeister) [reply tests/Feature/API/WpOrg/PluginUpdateCheckTest.php:1]: Hi.

That particular test was not supposed to be there. I just used to get the response from wp.org, but I forgot to remove it.

However, I do have other tests [like these](https://github.com/aspirepress/AspireCloud/blob/main/tests/Feature/API/WpOrg/Plugin_1_2_ControllerTest.php) where I hit the local API; those tests are not in the CLI and only run when explicitly invoked. 

Should I move those to Bruno as well?

[the remark this answers is not in the document]
Comment (chuckadams) [reply tests/Feature/API/WpOrg/PluginUpdateCheckTest.php:1]: Nah, those aren't making actual network requests, they're cool as they are.

[the remark this answers is not in the document]
Comment (Tmeister) [reply tests/Feature/API/WpOrg/PluginUpdateCheckTest.php:1]: The dummy test was removed.

**Review by chuckadams — APPROVED** [review]

**Pull request merged**